test/e2e/framework/pods.go

```diff
@@ -128,6 +128,9 @@ func (c *PodClient) mungeSpec(pod *api.Pod) {
 		if !TestContext.PrepullImages {
 			return
 		}
+		// Node e2e does not support the default DNSClusterFirst policy. Set
+		// the policy to DNSDefault, which is configured per node.
+		pod.Spec.DNSPolicy = api.DNSDefault
```

Comment:
We should move this before L128, right?
We want to do this no matter we prepull image or not.